File: app/src/main/java/com/tanim/androidbasetemplate/di/component/FragmentComponent.java
package com.tanim.androidbasetemplate.di.component;

import com.tanim.androidbasetemplate.di.module.FragmentModule;
import com.tanim.androidbasetemplate.di.scope.FragmentScope;

import dagger.Component;

@FragmentScope
@Component(modules = FragmentModule.class, dependencies = {AppComponent.class})
public
interface FragmentComponent {


}

File: app/src/main/java/com/tanim/androidbasetemplate/di/component/FragmentComponent.kt
package com.tanim.androidbasetemplate.di.component

import com.tanim.androidbasetemplate.core.home.HomeFragment
import com.tanim.androidbasetemplate.core.login.LoginFragment
import com.tanim.androidbasetemplate.di.module.FragmentModule
import com.tanim.androidbasetemplate.di.scope.FragmentScope
import dagger.Component

@FragmentScope
@Component(modules = [FragmentModule::class], dependencies = [AppComponent::class])
interface FragmentComponent {
    fun inject(loginFragment: LoginFragment)
    fun inject(homeFragment: HomeFragment)
} 